src/main/java/com/sleekbyte/tailor/Tailor.java

```diff
@@ -227,73 +246,129 @@ public static CommonTokenStream getTokenStream(File inputFile) throws IOExceptio
+                    TopLevelContext tree = optTree.get();
+                    List<SwiftBaseListener> listeners = createListeners(
+                        enabledRules,
+                        printer,
+                        tokenStream,
+                        constructLengths
+                    );
+
+                    walkParseTree(listeners, tree);
+
+                    try (FileListener fileListener =
+                             new FileListener(printer, inputFile, constructLengths, enabledRules)) {
+                        fileListener.verify();
                     }
-                    walker.walk(listener, tree);
-                }
-                try (FileListener fileListener = new FileListener(printer, inputFile, constructLengths, enabledRules)) {
-                    fileListener.verify();
-                }
 
-                numErrors += printer.getNumErrorMessages();
-                numWarnings += printer.getNumWarningMessages();
+                    numErrors += printer.getNumErrorMessages();
+                    numWarnings += printer.getNumWarningMessages();
+                } else {
+                    printer.printParseErrorMessage();
+                }
             }
+        } catch (IOException e) {
+            handleIoException(e);
+        } catch (ArgumentParserException e) {
+            handleCliException(e);
         }
+    }
+
+    /**
+     * Analyze files with SwiftLexer, SwiftParser and Listeners.
+     *
+     * @param fileNames List of files to analyze
+     * @throws ArgumentParserException if an error occurs when parsing cmd line arguments
+     * @throws IOException if a file cannot be opened
+     */
+    public static void analyzeFiles(Set<String> fileNames) throws ArgumentParserException, IOException {
+        numErrors = 0;
+        numWarnings = 0;
+
+        List<File> files = fileNames.parallelStream().map(File::new).collect(Collectors.toList());
+        ConcurrentMap<File, Optional<CommonTokenStream>> filesToTokenStreams =
+            files.parallelStream().collect(Collectors.toConcurrentMap(Function.identity(), Tailor::getTokenStream));
+        System.out.format("Analyzing %s:%n", pluralize(fileNames.size(), "file", "files"));
+        ConcurrentMap<File, Optional<TopLevelContext>> filesToTrees = files
+            .parallelStream()
+            .collect(Collectors.toConcurrentMap(Function.identity(),
+                file -> {
+                    Optional<TopLevelContext> tree = Optional.empty();
+                    try {
+                        tree = Tailor.getParseTree(filesToTokenStreams.get(file));
+                        System.out.print(".");
+                    } catch (ErrorListener.ParseException e) {
+                        System.out.print("S");
+                        numSkippedFiles++;
+                    }
+                    return tree;
+                }));
+        System.out.format("%n");
+        files.parallelStream().forEach(
```

Comment:
Using `parallelStream()` here sometimes results in unordered output of violation messages within a given file, and is fixed by switching to `stream()`. That change also seems to have minimal to no impact on performance.